roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -151,13 +160,12 @@ def available_images(self, images, registries, task_vars):
         ]
 
     def is_available_skopeo_image(self, image, registry, task_vars):
-        """Uses Skopeo to determine if required image exists in a given registry."""
+        """Uses Skopeo to determine if required image exists with a given registry."""
 
-        cmd_str = "skopeo inspect docker://{registry}/{image}".format(
-            registry=registry,
-            image=image,
-        )
+        # prepend registry only if image does not already have one
+        if image.count("/") < 2:
+            image = registry + "/" + image
```

Comment:
I mean, we can change this method signature to:

```python
def is_available_skopeo_image(self, image, run_command):
    result = run_command({"_raw_params": "skopeo inspect docker://" + image})
    return result.get("rc", 0) == 0 and not result.get("failed")
```

Where `run_command` knows about `task_vars` and `tmp` (or we pass in those and call `self.execute_module`).

This way `image` is whatever needs to go to the Skopeo call.

---

Aha, there have been no changes to the call sites of `is_available_skopeo_image`... so it means that we may be calling it multiple times with the same `image` and different `registry` and this new code simply makes us ignore the registry for certain images...

Maybe that was your intention?

The only thing to fix perhaps is to avoid calling it N times when the registry is pre-determined.